lib/neo4j/active_node/query/query_proxy.rb

```diff
@@ -191,6 +191,36 @@ def branch(&block)
           instance_eval(&block).query.proxy_as(self.model, identity)
         end
 
+        # Matches all nodes having at least a relation
+        #
+        # @example Load all people having a friend
+        #   Person.all.having_rel(:friends).to_a # => Returns a list of `Person`
+        #
+        # @example Load all people having a best friend
+        #   Person.all.having_rel(:friends, best: true).to_a # => Returns a list of `Person`
+        #
+        # @return [QueryProxy] A new QueryProxy
+        def having_rel(association_name, rel_properties = {})
+          association = model.associations[association_name]
+          fail "No such association #{association_name}" unless association
```

Comment:
This feels like we should have a `association` method which takes a `String` or `Symbol` as an argument and returns an association, but also would `fail` if the association doesn't exist.

Also, using `ArgumentError` would probably be appropriate